client/src/components/EmailFunction.tsx

```diff
@@ -0,0 +1,38 @@
+import { postRequestOptions } from "./Utils";
+
+const sendEmail = async (
+  body: string | undefined,
+  recipient: string | undefined
```

Comment:
Both the `body` and `recipient` shouldn't be of type undefined - you can change this to just string and if the param happens to be undefined, the err msg can come up quicker - bc i noticed while testing the err msg takes a while before actually popping up when I didn't input an email.